datafusion/core/tests/sqllogictests/test_files/aggregate_dict.slt

```diff
@@ -0,0 +1,83 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+
+#   http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+statement ok
+CREATE TABLE value(x bigint) AS VALUES (1), (2), (3), (4), (5);
+
+statement ok
+CREATE TABLE value_dict AS SELECT arrow_cast(x, 'Dictionary(Int64, Int32)') AS x_dict FROM value;
+
+query ?
+select x_dict from value_dict;
+----
+1
+2
+3
+4
+5
+
+query I
+select sum(x_dict) from value_dict;
+----
+15
+
+query R
+select avg(x_dict) from value_dict;
+----
+3
+
+query I
+select min(x_dict) from value_dict;
+----
+1
+
+query I
+select max(x_dict) from value_dict;
+----
+5
+
+query I
+select sum(x_dict) from value_dict where x_dict > 3;
+----
+9
+
+query R
+select avg(x_dict) from value_dict where x_dict > 3;
+----
+4.5
+
+query I
+select min(x_dict) from value_dict where x_dict > 3;
+----
+4
+
+query I
+select max(x_dict) from value_dict where x_dict > 3;
+----
+5
+
+query error DataFusion error: External error: Arrow error: Invalid argument error: RowConverter column schema mismatch, expected Int64 got Dictionary\(Int64, Int64\)
+select sum(x_dict) from value_dict group by x_dict % 2;
```

Comment:
Found another bug around Dictionary array with aggregate functions. This time `max`, `min` also have the issue. I only added tests and will address them separately later.

Reply:
![image](https://user-images.githubusercontent.com/490673/235898026-36175f4d-59e7-4876-8660-3a00a5b1143d.png)